tests/run/fused_cpdef.pyx

```diff
@@ -50,36 +46,45 @@ def test_fused_cpdef(printto=None):
     <BLANKLINE>
     B, x is long 2 long
     """
-    if printto is None:
-        printto = sys.stdout
-    func1[int](None, 2, printto)
-    func1[long](None, 2, printto)
-    func1(None, 2, printto=printto)
+    func1[int](None, 2)
+    func1[long](None, 2)
+    func1(None, 2)
 
-    print >> printto
+    print
 
-    pyfunc[cy.int](None, 2, printto=printto)
-    pyfunc(None, 2, printto=printto)
+    pyfunc[cy.int](None, 2)
+    pyfunc(None, 2)
 
-    print >> printto
+    print
 
-    A.meth[cy.int](A(), 2, printto=printto)
-    A.meth(A(), 2, printto=printto)
-    A().meth[cy.long](2, printto=printto)
-    A().meth(2, printto=printto)
+    A.meth[cy.int](A(), 2)
+    A.meth(A(), 2)
+    A().meth[cy.long](2)
+    A().meth(2)
 
-    print >> printto
+    print
 
-    B().meth(2, printto=printto)
+    B().meth(2)
 
 
 midimport_run = io.StringIO()
-midimport_run.write = lambda c: io.StringIO.write(midimport_run, unicode(c))
+try:
+    unicode
+except NameError:
+    pass
+else:
+    # Monkey-patch midimport_run.write to accept non-unicode strings under Python 2.
+    midimport_run.write = lambda c: io.StringIO.write(midimport_run, unicode(c))
+
+realstdout = sys.stdout
+sys.stdout = midimport_run
 
 try:
-    test_fused_cpdef(printto=midimport_run)
+    test_fused_cpdef()
 except Exception as e:
-    print >> midimport_run, repr(e)
+    print midimport_run, repr(e)
```

Comment:
```suggestion
    midimport_run.write(f"{e!r}\n")
```

That's something you can generally do here. No need for calling `print()`, just pass around `midimport_run.write` as an output function.